server/src/main/java/org/elasticsearch/script/field/HalfFloatDocValuesField.java

```diff
@@ -34,6 +34,11 @@ public HalfFloatDocValuesField(SortedNumericDoubleValues input, String name) {
         this.name = name;
     }
 
+    @Override
+    public Field<?> toScriptField() {
+        return this;
```

Comment:
There are a lot of these with the same implementation.  Does it make sense to have an `AbstractDocValuesScriptFieldSource` that implements `toScriptField`?

Reply:
Done.